pkg/issuer/acme/dns/azuredns/azuredns.go

```diff
@@ -49,7 +49,11 @@ func NewDNSProviderCredentials(environment, clientID, clientSecret, subscription
 		}
 	}
 
-	spt, err := getAuthorization(env, clientID, clientSecret, subscriptionID, tenantID, ambient)
+	if managedIdentityClientID != "" && managedIdentityResourceID != "" {
+		return nil, fmt.Errorf("managedIdentityClientID and managedIdentityResourceID can not be set at the same time")
+	}
```

Comment:
This should be enforced in the API validation. 

https://github.com/jetstack/cert-manager/blob/6f37d33c336c0d67c3c03b269fca114860e030f8/pkg/internal/apis/certmanager/validation/issuer.go#L319-L354

Reply:
I've moved validation to the (hopefully) right place